test/test_query.cpp

```diff
@@ -2011,6 +2009,19 @@ TEST(Query_SubtableExpression)
     tv = q.find_all();
     CHECK_EQUAL(tv.size(), 1);
     CHECK_EQUAL(tv.get_source_ndx(0), 1);
+
+    q = table->column<SubTable>(1).list<String>().begins_with(StringData("Str"));
+    tv = q.find_all();
+    CHECK_EQUAL(tv.size(), 3);
+    q = table->column<SubTable>(1).list<String>().ends_with(StringData("_8"));
+    tv = q.find_all();
+    CHECK_EQUAL(tv.size(), 1);
+    CHECK_EQUAL(tv.get_source_ndx(0), 2);
+    q = table->column<SubTable>(1).list<String>().begins_with(table->column<String>(2), false);
```

Comment:
Do you think bindings could ever desire the inverse of this? Namely: 
`q = table->column<String>(2).begins_with(table->column<SubTable>(1).list<String>());`
 (cc: @bdash )